src/core/common/settings.cpp

```diff
@@ -78,6 +78,13 @@ void SettingsBase::LogChildInfo(const char *aAction, const ChildInfo &aChildInfo
                   aChildInfo.GetMode(), aChildInfo.GetVersion());
 }
 
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+void SettingsBase::LogDadInfo(const char *aAction, const DadInfo &aDadInfo) const
+{
+    otLogInfoCore("Non-volatile: %s DadInfo {DadCounter:0x%2d}", aAction, aDadInfo.mDadCounter);
```

Comment:
```suggestion
    otLogInfoCore("Non-volatile: %s DadInfo {DadCounter:0x%02x}", aAction, aDadInfo.mDadCounter);
```
Sicne we have `0x` logging it as hex.
(or we should remove the `0x` and log as `%d`?

Reply:
good catch. will remove `0x`